lib/opam/opam_make_opam.ml

```diff
@@ -0,0 +1,163 @@
+open Core.Std
+open Core_extended.Std
+open Async.Std
+
+let opam_string_of_command_list
+  : String.t List.t -> String.t =
+  fun cmds ->  "[\n" ^
+               (List.fold ~init:""
+                  ~f:(fun acc el -> acc ^ "    [" ^ el ^ "]\n") cmds) ^
+               "]\n"
+let format_cmds
+  : default:String.t List.t -> String.t List.t -> String.t =
+  fun ~default cmds ->
+    match cmds with
+    | [] -> opam_string_of_command_list default
+    | _ -> opam_string_of_command_list cmds
+
+let create_build_cmds
```

Comment:
You are absolutely right. However, I prefer localizing the defaults to the function. the create_XXX_cmds only call the format_cmd with the right command. However, I find that much more readable then sticking the format commands into main function like you suggest.